cpp/src/parquet/column_writer.cc

```diff
@@ -1421,6 +1454,8 @@ class TypedColumnWriterImpl : public ColumnWriterImpl, public TypedColumnWriter<
   DictEncoder<DType>* current_dict_encoder_;
   std::shared_ptr<TypedStats> page_statistics_;
   std::shared_ptr<TypedStats> chunk_statistics_;
+  std::unique_ptr<SizeStatistics> page_size_statistics_;
```

Comment:
With regards to the comment at https://github.com/apache/arrow/pull/40594#discussion_r1856222399, I think it is much easier to use if `SizeStatistics::Make` returns an unique_ptr to match `std::shared_ptr<TypedStats>` above.